implementation/src/main/java/io/smallrye/config/ProfileConfigSourceInterceptor.java

```diff
@@ -0,0 +1,39 @@
+package io.smallrye.config;
+
+import javax.annotation.Priority;
+
+import org.eclipse.microprofile.config.Config;
+
+@Priority(600)
+public class ProfileConfigSourceInterceptor implements ConfigSourceInterceptor {
+    public static final String SMALLRYE_PROFILE = "smallrye.config.profile";
```

Comment:
Hum, isn't it risky? `config.profile` is somehow generic and we may clash with something already out there.

Reply:
That could be fine, especially if this were a constructor parameter.

Reply:
I can add the option to set that up as well.

Anyway, in other SmallRye projects, the namespace for config starts with `smallrye` shouldn't be consistent there?

Reply:
Perhaps we should have a configuration property to set the name of the configuration property... :slightly_smiling_face:

I guess it doesn't matter what the property name is as long as it can be set in the interceptor's constructor.

Reply:
Sure, I'll add it :)